src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
 
     /**
      * Method for getting key storing information about given on input column name with association. Can be override for customization needs.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @return string name 
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string name
      */
     protected function getColumnNameForAssociation(AuditLog $currentVersion)
     {
-        return $currentVersion->getTarget()->getTbl();
+        $joinTableName = $currentVersion->getTbl();
+        if (isset($this->cache['assocTable'][$joinTableName])) {
+            return $this->cache['assocTable'][$joinTableName];
+        }
+
+        foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
+            if (isset($assMapping['joinTable'])) {
+                $assJoinTable = $assMapping['joinTable']['name'];
+                $this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
+                if ($joinTableName === $assJoinTable) {
```

Comment:
Like before, we have here string comparison. Even by null identical equal comparison will work properly.

Reply:
according to our style guide / symfony style guide https://symfony.com/doc/current/contributing/code/standards.html#structure
 * Always use **identical comparison** *unless* you need type juggling

I do not need that `1 == '1'`